### DISCUSSION TITLE
recover topics from saved model

### DISCUSSION THREAD

**Discussion by aminathshausan** [MaartenGr/BERTopic#762] [Q&A] [answered]
Hi, 
I am new to using BerTopic. I have loaded a previously saved model. How do I get the _topics_ from the loaded model to perform the dynamic visualization in this step:
 topics_over_time = topic_model.topics_over_time(abstracts, topics, years)

Thanks

**Comment by MaartenGr** [comment, marked as answer]
It depends on the version of BERTopic you are using. If you are using v0.12, then you would not need to pass `topics` at all. However, if you are using an earlier version, I believe it would suffice to get those topics by running 

`topics = topic_model._map_predictions(topic_model.hdbscan_model.labels_)`

**Reply by aminathshausan** [reply]
Thanks, I have `version 0.11`.  I am using jupyter notebook with python 3.8. I tried to update the version by running 
`!pip install bertopic`
and then restarting the kernel,, but the imported package still shows `version 0.11`. 
Any help to upgrade to version 0.12 would be appreciated. 
Thanks

**Reply by MaartenGr** [reply]
To upgrade BERTopic, you will have to run `!pip install --upgrade bertopic` instead. When you run `!pip install bertopic`, it will not upgrade the version but only install BERTopic if it was not already installed.

**Reply by aminathshausan** [reply]
Thanks. I have upgraded BERTopic to version 0.12.0, and loading previously saved model gives the following error
`ModuleNotFoundError: No module named 'bertopic._ctfidf'`
What does this mean? And how do I fix this issue. 
Thanks

**Reply by MaartenGr** [reply]
You should make sure that the versions between saving and loading a model are exactly the same. The API can change between releases and as such an older version might not be supported by a newer version. It is similar to how [scikit-learn](https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations) handles this.

**Reply by aminathshausan** [reply]
Thanks. I did uninstall bertopic, reinstalled the latest version (0.12.0) and tried to import BERTopic using 
`from bertopic import BERTopic`
I am now getting the following Type error:
`TypeError: __init__() got an unexpected keyword argument 'cachedir'`
What does this mean and how do I fix it please.  

I am getting the same Type error when installing version 0.11.0 too
Thanks

**Reply by MaartenGr** [reply]
This is a known issue and you can find more information about that [here](https://github.com/MaartenGr/BERTopic/issues/730) and [here](https://github.com/MaartenGr/BERTopic/issues/770).

**Reply by aminathshausan** [reply]
Thanks, I could fix the error following the links, and successfully imported BERTopic. But loading the saved model now gives the following error with BERTopic version 0.11.0. 

`AttributeError: Can't get attribute '_rebuild_function' on <module 'numba.core.serialize' from '/Users/.../opt/anaconda3/lib/python3.8/site-packages/numba/core/serialize.py'>`

Any help please
Thanks

**Reply by MaartenGr** [reply]
Did you make sure that the environment in which you saved the model is *exactly* the same as the environment in which you loaded the model? Seeing as you did not have this issue saving the model, most likely something has changed between environments. I would advise starting from a completely fresh environment, making sure you keep track of dependencies and re-train your model. That way, by tracking the package versions in your environment, you prevent most of these issues.

**Reply by aminathshausan** [reply]
Thanks- that worked
